### PR TITLE
Smoothly scale the Pesto logo as the app bar resizes.

### PR DESCRIPTION
**Pull request by mpcomplete**
Also update the assets version to pull in better quality logo images.

BUG=https://github.com/flutter/flutter/issues/4407

### CODE REVIEW

**Comment by abarth** [comment]
LGTM

File: examples/flutter_gallery/lib/demo/pesto_demo.dart [hunk 1]
```diff
@@ -108,12 +111,18 @@ class _PestoDemoState extends State<PestoDemo> {
           double bestHeight = _kLogoImages.keys.lastWhere(
             (double height) => appBarHeight >= height
           );
+          // Extra padding. Calculated to give about 16px on the bottom for the
+          // `small` logo at its native size, and 30px for the `medium`.
+          double extraPadding = min(.19*appBarHeight + 5.4, 40.0);
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/pesto_demo.dart]: Include the leading zero, and put spaces around operators.

File: examples/flutter_gallery/lib/demo/pesto_demo.dart [hunk 2]
```diff
@@ -108,12 +111,18 @@ class _PestoDemoState extends State<PestoDemo> {
...
           return new Padding(
-            padding: new EdgeInsets.only(top: statusBarHeight),
+            padding: new EdgeInsets.only(
+              top: statusBarHeight + extraPadding*.5,
```
Comment (Hixie) [inline examples/flutter_gallery/lib/demo/pesto_demo.dart]: here too

**Comment by Hixie** [comment]
Travis' anger is because you have to update the version number of the gallery assets package in three places now. I recommend leaving a comment in each of those places pointing to the other two.

**Pull request merged**